### ISSUE TITLE
Slider

### ISSUE THREAD

**Issue by ulde01** [sebgroup/design-library#18]

**Comment by henrikhoglundseb** [comment]
I have some suggestions on a iterated design and more option like a range-selector. @rey have the need in her project for that component. 

She might take this further. 

<img width="503" alt="Screenshot 2019-04-24 at 11 26 47" src="https://user-images.githubusercontent.com/46597007/56648794-37d60580-6684-11e9-81fd-44289a54e9b2.png">

**Comment by rey** [comment]
?

**Comment by henrikhoglundseb** [comment]
hihi, wrong rey :)

**Comment by TheresaHedlund** [comment]
Hi,
Maybe we can just do minor adjustments on the one we have. Some suggestions;

- make the drag thing less wide
- adding +/ - for changing values step by step
- possibility to add max/min values

Quick sketch:
![image](https://user-images.githubusercontent.com/47220738/59195362-0e1d6300-8b8c-11e9-82e7-c1e925d6aead.png)

**Comment by rey** [comment]
> adding +/ - for changing values step by step

I'm not sure whether the +/- add much value? Though it would depend on the scale of numbers that's being used e.g. 1 ➡️ 5 vs. 1 ➡️ 100.

**Comment by TheresaHedlund** [comment]
> > adding +/ - for changing values step by step
> 
> I'm not sure whether the +/- add much value? Though it would depend on the scale of numbers that's being used e.g. 1 ➡️ 5 vs. 1 ➡️ 100.

Could be optional. Also the values under slider might not always be needed.

**Comment by henrikhoglundseb** [comment]
I agree with Rey. I think the + and - add more complexity and removes some off the slider behavior. 

I like the possibility to add min and max values and to add scaling, if its needed. 

I would like to make the slider buttons round. Having square or rectangular slider-buttons just feels wrong. Its more comon to have them round.

**Comment by conandx** [comment]
We have previously discussed to have the possibility to let the slider snap to set intervals, for intance even 1000 or similar (depending on scale). 
Very many of the features of a slider should be optional so it's possible to set it up for each occasion.

**Comment by EldRoos** [comment]
React components now have an updated version of slider with hover and focus states. Henrik and Ulrika was in on this.

We have choosen version 1 of these for focus:
![SliderFocus](https://user-images.githubusercontent.com/14058437/68948933-45d2ab80-07b9-11ea-8622-46c64f5eb86e.png)

Hover animates to a bit larger:
![SliderHover](https://user-images.githubusercontent.com/14058437/68930666-26268d80-078f-11ea-98eb-c191acbec9a2.png)

Negative slider:
![SliderMid](https://user-images.githubusercontent.com/14058437/68950686-f8583d80-07bc-11ea-9d25-77c5efdab216.png)

---

Me and Jessi have taken the torch and made a few suggestions for new graphics. Where the 3rd from bottom is the favorite:
![Sliders](https://user-images.githubusercontent.com/14058437/68930451-b4e6da80-078e-11ea-8a38-795975a9165d.png)

Note: but this small one needs to have a bigger hitbox for big fingers.

**Comment by TheresaHedlund** [comment]
> Me and Jessi have taken the torch and made a few suggestions for new graphics. Where the 3rd from bottom is the favorite

Yay! Will work so much better when it's smaller :D 
I also like the classic round ones. They feel very touch/drag friendly, since you're so used to them.

**Comment by ulde01** [comment]
Nice!

**Comment by ghost** [comment]
I hear you @TheresaHedlund But think the square one is more in line with our other button like components. We need to make sketches on more features right? But let´s start with this small update?
Anyone else have comment? 
/Jessi

**Comment by henrikhoglundseb** [comment]
I cast my vore on circular “sliders” some components just look better, I understand that we are the square brand.

Just want to add my comment.

Sent from my iPhone

On 20 Nov 2019, at 13:15, JessiNygrenWalhed <notifications@github.com> wrote:

﻿

I hear you @TheresaHedlund<https://github.com/TheresaHedlund> But think the square one is more in line with our other button like components. We need to make sketches on more features right? But let´s start with this small update?
Anyone else have comment?
/Jessi

**Comment by TheresaHedlund** [comment]
> I hear you @TheresaHedlund But think the square one is more in line with our other button like components. We need to make sketches on more features right? But let´s start with this small update?
> Anyone else have comment?
> /Jessi

Yeah, I guess. Unless you want to compare with a toggle. ;)
![image](https://user-images.githubusercontent.com/47220738/69253792-7baad200-0bb5-11ea-9031-4db6f7519d0c.png)

Square is fine with me. As long as they get smaller!

**Comment by conandx** [comment]
I absolutely think we should keep it simple. I've seen both square and round controlls, and in that context I beleive the square is more appropriate for our design scheme.

**Comment by henrikhoglundseb** [comment]
Should we change the toggle switch also then ?

________________________________

**Comment by EldRoos** [comment]
We will have a meting about the new design soon, who want to join in addition to me - Leopold Roos, and Jessi Nygren Walhed?

What features are missing in current design? Except the range feature.

Also: Yea I think the toggle should also be squarish :). Good to be a bit different to others.

**Comment by henrikhoglundseb** [comment]
I would like to join. Im working on a new small project that is based on sliders

Sent from my iPhone

On 5 Dec 2019, at 10:12, EldRoos <notifications@github.com> wrote:

﻿
CAUTION: This email originated from outside of the organization.
________________________________
Do not click links or open attachments unless you recognize the sender and know the content is safe.


We will have a meting about the new design soon, who want to join in addition to me - Leopold Roos, and Jessi Nygren Walhed?

What features are missing in current design? Except the range feature.

**Comment by EldRoos** [comment]
We had a meeting about Slider component and decided to mostly change the visual this time. Extra features can wait until someone actually needs them – agile way.

•	Lighter gray on the bar:en (gray-400 #cecece instead of gray-300 #dedede).
•	Bar should have rounded edges (4px border-radius instead of none).´
•	Blue bar should start from 0 or other defined in midpoint (not all to the left like before).
•	Thumb should be smaller square (24x24px instead of 48x32px) with round borders (4px border-radius). No lines or chevrons inside.
•	Focus as the earlier prototype – lightblue thick line with (3px with 4px border-radius) on thumb.
•	Hover as the earlier prototype – increase size animation of 5% on thumb.
•	Tooltip should have no border-radius on the black square (instead of 4px).
•	Tooltip should support thousand deliminator and unit.

Range slider have to be defined by the one that needs it. But should follow same style.

![Slider](https://user-images.githubusercontent.com/14058437/70633923-9e1e9100-1c31-11ea-9132-c461e1bdb02d.png)

**Comment by ulde01** [comment]
@EldRoos is updating the design after the design meeting from before Christmas. Updated design and description to follow.

**Comment by ulde01** [comment]
Denna komponent är inte uppdaterad i Sketch. Leopold har uppdaterat i React.  24 x 24 thumb, chevron är borttagna, 4 px height på linjen som även har rundade hörn. På hover förstoras tummen och på focus så blir det en ram runt tummen.
